main.py

```diff
@@ -0,0 +1,364 @@
+    file_path = f"{RAW_PATH.format(platform)}{filename}"
+
+    # Checks if the file already exists
+    if not os.path.isfile(file_path):
+        print("No file found, creating it!\n-----------------")
+        r = requests.get(URL.format(platform))
+        r.raise_for_status()
+        data = r.json()
+        with open(file_path, "w") as outfile:
+            json.dump(data, outfile, indent=4)
+    else:
+        print("File already exists!\n-----------------")
+
+
+def process_data(platform):
+    '''Calls all of the processing functions for every raw file'''
+
+    # Goes over every file in the raw folder and processes the data for it
+    for filename in os.listdir(RAW_PATH.format(platform)):
+
+        print(f"\nProcessing {filename}\n")
+
+        date = filename.rstrip(".json")[-8:]
+        raw_file_path = f"{RAW_PATH.format(platform)}{filename}"
+        edit_file_path = f"{EDIT_PATH.format(platform)}{filename}"
+
+        with open(raw_file_path, "r") as raw:
+            raw_data = json.load(raw)
+            sales = sale_calc(raw_data)
+            raw_dict = create_dict(raw_data, 1)
+            del raw_data
+
+        # Checks if the raw data is in the TOTAL file and returns the total dict
+        total_dict = total_json(platform, raw_dict, date, sales)
+
+        if os.path.isfile(edit_file_path):
+            print(f"/edited/{filename} Aleady exists!")
+        else:
+            comparison(platform, raw_dict, total_dict, sales, filename)
+
+
+def sale_calc(data):
+    '''Calculates the number of sales'''
+    print("Calculating the total sales number!")
+    min_price = []
+    max_price = []
+    pop = []
+    for d in data:
+        min_price.append(d["min"])
+        max_price.append(d["max"])
+        pop.append(d["pop"])
+
+    min_pop = min(pop)
+    pop_ind = pop.index(min_pop)
+    if min_price[pop_ind] == max_price[pop_ind]:
+        trades = 100 / min_pop
+        return trades
+    else:
+        return None
+
+
+def total_json(platform, raw_dict, date, sales):
+    '''Creates the total files'''
+    try:
+        with open(TOTAL_FILE_PATH.format(platform, platform), "r") as file:
+            total_file = json.load(file)
+            total_dict = create_dict(total_file, 3)
+            dates = total_file[0]
+            del total_file
+        print(f"Found TOTAL_{platform} adding data!")
+
+        if date in dates:
+            print(f"Data from {date} already in TOTAL!")
+            return total_dict      # Will stop the function to prevent the unneeded saving of the same data
+        else:
+            dates.append(date)
+
+            for key in list(raw_dict.keys()):
+                if raw_dict[key]["median"] == 0:
+                    median_count = 0
+                else:
+                    median_count = 1
+
+                raw = raw_dict[key]
+                # File exists riven already in file
+                if key in total_dict:
```

Comment:
The `if-else` block here can be optimized a bit. Basically we initialize it first if it does not exist. Then we update it.

```py
# so if it does not exist, we initialize it
if key not in total_dict:
  total['itemType'] = raw['itemType']
  total['total_min'] = 0  # notice how its zero

# when the code reaches here, we can always safely update!
total["total_min"] += raw["min"]
```

Reply:
Good catch, changed it. I also removed the `total["itemType"] = raw["itemType"]` lines since those values will not change and are already in the file or are created couple lines above.